### PR TITLE
DF-371-Simplify-upgrade-matrix-in-PGD-4-and-PGD-5-docs

### PR DESCRIPTION
**Pull request by djw-m**
## What Changed?

Matrices are now tables mapping version to available upgrades reducing the inpage complexity for reader and authors.

### CODE REVIEW

File: product_docs/docs/pgd/5/upgrades/upgrade_paths.mdx
Comment (djw-m) [reply product_docs/docs/pgd/5/upgrades/upgrade_paths.mdx]: Fixed

File: product_docs/docs/pgd/4/upgrades/upgrade_paths.mdx
Comment (djw-m) [reply product_docs/docs/pgd/4/upgrades/upgrade_paths.mdx]: Fixed

**Review by mwanner — APPROVED** [review]
SGTM

File: product_docs/docs/pgd/4/upgrades/upgrade_paths.mdx
Comment (LenzGr) [inline product_docs/docs/pgd/4/upgrades/upgrade_paths.mdx:7]:
```suggestion
Beginning with version 4, EDB Postgres Distributed has adopted [semantic versioning](https://semver.org/). All changes within the same major version will be backward compatible, lowering the risk when upgrading and allowing you to choose any later minor or patch release as the upgrade target. You can upgrade from any version 4.x release to a later 4.x release.
```

File: product_docs/docs/pgd/5/upgrades/upgrade_paths.mdx
Comment (LenzGr) [inline product_docs/docs/pgd/5/upgrades/upgrade_paths.mdx:11]: Is "Generally, we recommend" strong enough wording here? Didn't we want to make this a requirement before upgrading to the next major version?

Comment (djw-m) [reply product_docs/docs/pgd/5/upgrades/upgrade_paths.mdx:11]: There may be circumstances where that is not the best path. This wording allows for exceptions, without having contradictory content in the docs.

**Review by LenzGr — DISMISSED** [review]
Looks good to me; let's wait for @kelpoole review/approval.

File: product_docs/docs/pgd/4/upgrades/upgrade_paths.mdx
Comment (mwanner) [inline product_docs/docs/pgd/4/upgrades/upgrade_paths.mdx:27]: I checked the release tags specifically and figured this table matches more the "has been released at the same time" rather than what the upgrade capabilities actually are. The "possible upgrade paths" deviate slightly and should look more like:

To me, it looks more like:
```
| From version | To version |
| ---- | -- |
| 3.7.15 | 4.0.2 or later |
| 3.7.16 | 4.1.0 or later |
| 3.7.17 | needs to be upgraded to a newer 3.7.x version, first |
| 3.7.18 | 4.2.1 or later |
| 3.7.19 | 4.2.2 or later |
| 3.7.20 | 4.3.0 or later |
| 3.7.21 | 4.3.1 or later |
| 3.7.22 | 4.3.2 or later |
```

File: product_docs/docs/pgd/5/upgrades/upgrade_paths.mdx
Comment (mwanner) [inline product_docs/docs/pgd/5/upgrades/upgrade_paths.mdx:21]: I'm still not quite clear what `4.3.0-1` refers to. In the packaging repo, I see all of these tags:
* `REL_4_3_0-1`
* `REL_4_3_0-2`
* `REL_4_3_0-3`

...there's no `REL_4_3_0-0`, though. So ... what's the difference between `4.3.0` vs `4.3.1-1`?

Clearly, BDR version 5.0.1 provides an upgrade script from 4.3.0, so for that reason I'd conclude that "4.3.0" (any packaging revision) is upgradable to "5.0.1 or later" (but then scratch any "4.3.0-1" rows).

File: product_docs/docs/pgd/5/upgrades/upgrade_paths.mdx
Comment (mwanner) [inline product_docs/docs/pgd/5/upgrades/upgrade_paths.mdx:23]: Pretty much the same here, I'd just scratch the `4.3.1-1` line.

**Review by mwanner — CHANGES_REQUESTED** [review]

File: product_docs/docs/pgd/5/upgrades/upgrade_paths.mdx
Comment (mwanner) [reply product_docs/docs/pgd/5/upgrades/upgrade_paths.mdx:21]: Okay, that got clarified via https://github.com/EnterpriseDB/docs/pull/4755. That however means all the `+p1` versions match their original one, because the only component that really matters for upgrades is BDR.

File: product_docs/docs/pgd/5/upgrades/upgrade_paths.mdx
Comment (djw-m) [reply product_docs/docs/pgd/5/upgrades/upgrade_paths.mdx:23]: So just pull the +p releases lines leaving only the BDR version? I asm as 4.3.0+p1 goes the 5.0.1 not 5.0.0

File: product_docs/docs/pgd/5/upgrades/upgrade_paths.mdx
Comment (djw-m) [reply product_docs/docs/pgd/5/upgrades/upgrade_paths.mdx:21]: So for the entry 

| 4.3.1 | 5.1.0 or later |
| 4.3.1+p1 | 5.2.0 or later |
| 4.3.2 | 5.2.0 or later |

Do you want the +p1 row there deleted? or retained? Or is that another error in the upgrade data?

File: product_docs/docs/pgd/5/upgrades/upgrade_paths.mdx
Comment (djw-m) [reply product_docs/docs/pgd/5/upgrades/upgrade_paths.mdx:23]: Like I said in the other change, the 4.3.1 -> 4.3.1+p1 bumps the target version from 5.1.0 to 5.2.0

Is that (a) correct and (b) if it is, shouldn't we keep it in?

File: product_docs/docs/pgd/4/upgrades/upgrade_paths.mdx
Comment (djw-m) [reply product_docs/docs/pgd/4/upgrades/upgrade_paths.mdx:27]: Switched to the given list

**Comment by mwanner** [comment]
> Looks good to me; let's wait for @kelpoole review/approval.

I suggest we move ahead with this, as Kelly was out sick and may have a backlog of things. What we have clearly is an improvement. It's more accurate and simpler to understand. And we can always refine.

@djw-m Please resolve the conflicts, first.

**Review by mwanner — DISMISSED** [review]

**Comment by djw-m** [comment]
Conflict is resolved. Could I have one fresh LGTM?

**Comment by djw-m** [comment]
Continues in #4766

**Review by mwanner — APPROVED** [review]

**Pull request closed without merging**
